src/action.ts

```diff
@@ -95,7 +96,13 @@ export function handleActionConfig(
         return;
       }
       const [domain, service] = actionConfig.service.split(".", 2);
-      hass.callService(domain, service, actionConfig.service_data);
+      callService(
+        hass.connection,
+        domain,
+        service,
+        actionConfig.service_data,
+        actionConfig.service_target
+      );
```

Comment:
I knew this will go the wrong way. Initially I had it like that, but then decided to use this as I thought there's a pattern to separate the additional attributes related to a service call. I'll change it back.

Reply:
Using `hass.callService` does not work though as it comes from `custom-card-helpers` and does not support this function signature. It defines `hass` with it's own `HomeAssistant` interface which is not compatible with the default import from `home-assistant-js-websocket` either.

Try replacing the imports section of `action.ts` with this:

```ts
import {
  fireEvent,
  forwardHaptic,
  navigate,
  toggleEntity
} from "custom-card-helpers";
import { ButtonActionConfig, ButtonConfig } from "./types";
import { callService } from "home-assistant-js-websocket";
import homeAssistantWS from "home-assistant-js-websocket";

type HomeAssistant = typeof homeAssistantWS;
```

You should then see something like this:

![image](https://user-images.githubusercontent.com/9029939/147074937-70775c6d-4b2e-42af-b668-f3c23ffc15af.png)

Reply:
Wow, well thankfully we were not the only ones surprised by this and support has already been added for this (see my comment below).